source/java/src/org/lucee/extension/cache/eh/util/EHCacheClassLoader.java

```diff
@@ -0,0 +1,135 @@
+package org.lucee.extension.cache.eh.util;
+
+import java.lang.reflect.Method;
+
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+
+import lucee.loader.engine.CFMLEngine;
+import lucee.loader.engine.CFMLEngineFactory;
+import lucee.runtime.config.Config;
+
+import lucee.commons.io.log.Log;
+
+
+/**
+ * A special class loader that first attempts to load class from the Lucee Environment,
+ * but that can fail when RMI classes are requested via a spawned thread, so when
+ * the class cannot be located in the Lucee Environment, it searches all the Felix
+ * bundles for the requested class. This will prevent issues with issues trying
+ * to load classes like net.sf.ehcache.distribution.RMICachePeer_Stub. 
+ *
+ * @author Dan G. Switzer, II
+ * @version 1.0
+ */
+public class EHCacheClassLoader extends ClassLoader {
+	private Config config;
+	private ClassLoader LuceeEnvClassLoader;
```

Comment:
var should start lower case, so "luceeEnvClassLoader", also i don't like to have used the  word "Lucee" in a variable name, it is never really benefital.

Reply:
I totally understand naming stuff `Lucee` usually isn't helpful. I used that name because I was trying to make it clear that this was the class loader used by Lucee internally. However, I've renamed the variable to `coreEnvClassLoader`. I'm open to a better name if you have one.